orga/app/components/auth/register-form.js

```diff
@@ -2,106 +2,132 @@ import { action } from '@ember/object';
+import get from 'lodash/get';
 
 export default class RegisterForm extends Component {
   @service session;
   @service store;
   @service intl;
 
-  validation = {
-    firstName: new InputValidator(
-      isStringValid,
-      this.intl.t('pages.login-or-register.register-form.fields.first-name.error')
-    ),
-    lastName: new InputValidator(
-      isStringValid,
-      this.intl.t('pages.login-or-register.register-form.fields.last-name.error')
-    ),
-    email: new InputValidator(isEmailValid, this.intl.t('pages.login-or-register.register-form.fields.email.error')),
-    password: new InputValidator(
-      isPasswordValid,
-      this.intl.t('pages.login-or-register.register-form.fields.password.error')
-    ),
-    cgu: new InputValidator(Boolean, this.intl.t('pages.login-or-register.register-form.fields.cgu.error')),
-  };
-
-  user = null;
   @tracked isLoading = false;
-  @tracked isPasswordVisible = false;
-
-  get passwordInputType() {
-    return this.isPasswordVisible ? 'text' : 'password';
-  }
-
-  constructor() {
-    super(...arguments);
-    this.user = this.store.createRecord('user', {
-      lastName: '',
-      firstName: '',
-      email: '',
-      password: '',
-      cgu: false,
-    });
-  }
+  @tracked firstName = null;
+  @tracked lastName = null;
+  @tracked email = null;
+  @tracked password = null;
+  @tracked passwordValidationMessage = null;
+  @tracked emailValidationMessage = null;
+  @tracked firstNameValidationMessage = null;
+  @tracked lastNameValidationMessage = null;
+  @tracked cguValidationMessage = null;
+  @tracked errorMessage = null;
 
   @action
   async register(event) {
     event.preventDefault();
-    if (this._isFormInvalid()) {
+    this.errorMessage = null;
+    if (!this._isFormValid()) {
       return;
     }
     this.isLoading = true;
     try {
-      await this.user.save();
-    } catch (e) {
+      await this.store
+        .createRecord('user', {
+          lastName: this.lastName,
+          firstName: this.firstName,
+          email: this.email,
+          password: this.password,
+          cgu: true,
+        })
+        .save();
+    } catch (response) {
       this.isLoading = false;
-      return this._updateInputsStatus();
+
+      const status = get(response, 'errors[0].status');
+
+      if (status === '422') {
+        this.errorMessage = this.intl.t('pages.login-or-register.register-form.errors.email-already-exists');
+      } else {
+        this.errorMessage = this.intl.t('pages.login-or-register.register-form.errors.default');
+      }
+      return;
     }
 
     await this._acceptOrganizationInvitation(
       this.args.organizationInvitationId,
       this.args.organizationInvitationCode,
-      this.user.email
+      this.email
     );
-    await this._authenticate(this.user.email, this.user.password);
+    await this._authenticate(this.email, this.password);
 
-    this.user.password = null;
+    this.password = null;
     this.isLoading = false;
```

Comment:
perso je trouve que de mettre le return ; dans le catch apporte a confusion, d'ou le fait que tu l'ai loupe la premiere fois
je serai plus a mettre tout ce code dans le try, a la suite du save et mettre this.isLoading = false dans un finally au lieu de le repeter 2 fois